### ISSUE TITLE
✅ Toggle checkboxes in description on click

### ISSUE THREAD

**Issue by neerro** [stefan-niedermann/nextcloud-deck#756]
**Describe the bug**
I've attached screenshot for clear understanding. There are no way for easy using markdown check-list in description box inside of tasks. 
First of all we may observe some point before starting markdown code on the first line.
The second one I have no possibility to  mark the tasks in check-list just to put on click, for this option I should manually put X inside [ ]. For example in Web UI I have a possibility marking finalized tasks just put on click. 

**To reproduce**
Issue is reproducible with an account from [try.nextcloud.com](https://try.nextcloud.com):

- [ ] Yes
- [x] No

**Expected behavior**
I would to get a possibiluty marking tasks which were done by put on click. Also it would be greate to add feature when  tasks in check-list will be done to indicate like this one: ~~Text of the task~~

**Screenshots**
![Screenshot_20201212_001349_it.niedermann.nextcloud.deck.play.jpg](https://user-images.githubusercontent.com/75830569/101955864-a546f900-3c0f-11eb-8ef7-5d98f0409abd.jpg)

**Versions**
 - Nextcloud: latest
 - Nextcloud Deck: latest
 - Nextcloud Android: latest
 - Nextcloud Android Deck: latest

**Smartphone (please complete the following information):**
- Device: Honor 10 lite
- Android-Version: 10
- App-Store:
  - [x] Google Play Store
  - [ ] Google Play Store (Beta channel)
  - [ ] F-Droid
  - [ ] Huawei AppGallery

**Are you using LDAP?**
- [ ] Yes
- [x] No

**Comment by stefan-niedermann** [comment]
Well, actually it's not a bug but a feature request. 😉 

We replaced the complete markdown editor recently and released version `1.13.1` at the Play Store Beta channel - this will bring a few enhancements to the editor but not solve your issue yet (toggle checkboxes by click).

I will keep this issue open as a feature request, so we don't forget it. 🙂 

> First of all we may observe some point before starting markdown code on the first line.

Unfortunately I do not understand this

> Also it would be greate to add feature when tasks in check-list will be done to indicate like this one: ~~Text of the task~~

This propably won't happen. It changes the semantic of the content and while it might fit in most scenarios, it won't fit for *each* scenario. I recommend to explicitly use the `~~strike~~` syntax for that.

**Comment by neerro** [comment]
Thank you Stefan for your fast response and your correction. Yes of course it should be feature request :)

> "Unfortunately I do not understand this"
> Also it would be greate to add feature when tasks in check-list will be done to indicate like this one: ~~Text of the task~~

So, I mean when we mark the task as done (for example: clicked on toggle checkbox) the text is automatically become strikethrough (as: - [x] ~~Text~~). For example this feature has implemented in Trello. From my lowly point of view many functions could be taken from Trello and will implemented in Deck for best user practice.

> This propably won't happen. It changes the semantic of the content and while it might fit in most scenarios, it won't fit for each scenario. I recommend to explicitly use the ~~strike~~ syntax for that.
Yes, but it not so convenient for daily quick using.

**Comment by stefan-niedermann** [comment]
Step by step 🙂 As i said, i agree that toggling checkboxes is a must have feature and we will support it at some point in the future.

But as a simple example where a ~~striked~~ text will completely change the semantic, take this description as an example:

```
Title: BBQ next friday
Description:
We need to find out how much sausages we will have.
Neighbours who will bring sausages:

- [x] Thomas
- [ ] Sarah
- [x] Bernhard
```
In this case actuall `Sarah` needs to be displayed as ~~Sarah~~, so it ain't that easy. I will have a talk with a few designers about this topic, but for now we will only commit to implement toggleable checkboxes via click.

**Comment by neerro** [comment]
> Title: BBQ next friday
Description:
We need to find out how much sausages we will have.
Neighbours who will bring sausages:
- [x] Thomas
- [ ] Sarah
- [x] Bernhard
In this case actuall Sarah needs to be displayed as Sarah, so it ain't that easy. I will have a talk with a few designers about this topic, but for now we will only commit to implement toggleable checkboxes via click.

No. It's actually not. For this case we have to separate Sara from this list. For example (I'd made a little bit different. See below):

Title: BBQ next friday
Description:
We need to find out how much sausages we will have.
Neighbours who will bring sausages:
- Thomas
- Bernhard
Neighbours who will bring souse for sausages:
- Sarah

The example where we can use a ~~strike~~ text is:

What I have to do when the next morning will come? (Let's imagine that some tasks already done and how it have to be look)
- [x] ~~woke up~~
- [x] ~~stand up from bed~~
- [] wash oneself
- [] make coffee
- [] ... etc.

That is my case what do I mean and how it should be work for ~~strike~~ text (in automation mode when we click on toggle box).

This is my point of view how those scenarios should be working.

**Comment by desperateCoder** [comment]
If I may add my own two cents:
> This is my point of view how those scenarios should be working.

@neerro: Yep, thats **your** point of view. I bet if you ask ten people, you get twelve points of views.

>  and while it might fit in most scenarios, it won't fit for each scenario.

@stefan-niedermann: Yep, not all, but **the most**.

IMHO you both got absolutely valid points. In cases like these I'm about to say: let the user decide. Make it a setting to toggle the behavior. Since I am just the backend-dude here and @stefan-niedermann is the master of UI/UX you should know my opinion isn't really important here. We also want to integrate fine into the Nextcloud ecosytem and match the most expectations of our users, we might not be able to fit **all** of them.

> I will have a talk with a few designers about this topic, but for now we will only commit to implement toggleable checkboxes via click.

@neerro so you'll get at least something. Is it ok for you to take a step back and let @stefan-niedermann go through it step-by-step? I promise we will consider this and maybe I can convince him to make it at least toggleable, while we can't promise the Nextcloud UX designers will like the idea. And even if they do, there are some technical obstacles for this, since we just integrated and enhanced a brand new Markdown-engine. So please be patient, we do this all in our spare time and got a bunch of other issues to solve, so we concentrate on the most urgent ones, while this one isn't the game changer (for now!).

Still, I love configurable software @stefan-niedermann :stuck_out_tongue_winking_eye:

**Comment by politas** [comment]
So what's the current state of markdown support supposed to be in the mobile app? I've got the latest version from the play store, and it seems to not be rendering any markdown elements.

**Comment by stefan-niedermann** [comment]
@politas i really don't get what you mean.

Sample screenshots directly taken from version `1.13.4` which show the support for
- headlines
- lists
- checkboxes
- quotes
- syntax highlighting and
- tables

in
- edit mode (left) and
- preview mode (right) (clicking on the :eye: eye icon in the top right of the description):

| 1 | 2 |
| --- | --- |
| ![grafik](https://user-images.githubusercontent.com/4741199/103826839-ef3cc680-5077-11eb-86c2-68088842a3ff.png) | ![grafik](https://user-images.githubusercontent.com/4741199/103826856-f9f75b80-5077-11eb-9ce4-5bf121cc45b5.png) |
| ![grafik](https://user-images.githubusercontent.com/4741199/103826983-39be4300-5078-11eb-8957-79470a47a970.png) | ![grafik](https://user-images.githubusercontent.com/4741199/103826993-43e04180-5078-11eb-9867-fa71a4cd2bcd.png) |

So i suppose you
- don't comment to this specific issue about toggling checkboxes in the preview mode
- provide a ***tiny littly*** more information about what *exactly* you are missing (except toggling checkboxes, because we already know that and are actually working on it :wink: )
- provide more information about your environment, because "latest version from the play store" is not very specific. You can find this information in the "About" section of the sidebar
- open a new issue for your points, *with* example markdown, so we can separate the issues and work on them without having one huge issue which will never get completed

Have a nice evening. :wave:

**Comment by politas** [comment]
Ah, I guess my problem is that I was expecting to open cards in view mode by default, and didn't notice that tiny eye symbol.

**Comment by stefan-niedermann** [comment]
Glad you found it now 🙂. We will think about adding a hint and a setting for open always in view mode or rember the last decision in the future.

**Comment by dudeofawesome** [comment]
Opening in view mode would be nice, given that is the default view for the web app as well.

**Comment by stefan-niedermann** [comment]
Checkboxes can be toggled within view mode starting with `1.15.0`.

> Opening in view mode would be nice, given that is the default view for the web app as well.

Please open a separate issue about that topic. We will need some UX input here, because valid options are
- Always open in view mode
- Always open in edit mode
- Remember last state per card
- Remember last state per board
- Remember last state globally

Those are too many options to maintain and i'd like to pick some useful defaults :) -> Discussion needed.
